### PR TITLE
Refactor controller to use Automation API

### LINKED ISSUE #85: Refactor controller to use Automation API
### Problem description

Refactor the logic in the controller to use [Automation API](https://github.com/pulumi/pulumi/pull/4977) for stack operations.

Generally refactor:
- [x] Project setup & Create stack
- [x] Update stack
- [x] Refresh stack
- [x] Destroy stack

### PR DESCRIPTION
**Pull request by metral**
### Proposed changes

Refactor controller to use Automation API

### Related issues (optional)

Closes #85

### CODE REVIEW

File: pkg/controller/stack/stack_controller.go
Comment (lblackstone) [inline pkg/controller/stack/stack_controller.go:456]:
```suggestion
			cmd.Env = append(cmd.Env, strings.Sprintf("%s=%s", k, v))
```

File: pkg/controller/stack/stack_controller.go
Comment (lblackstone) [inline pkg/controller/stack/stack_controller.go:698]:
```suggestion
		var value apiextensionsv1.JSON
```

File: pkg/controller/stack/stack_controller.go
Comment (lblackstone) [inline pkg/controller/stack/stack_controller.go:738]:
```suggestion
	substr := permalinkStart[:end[1]-1]
```

**Review by lblackstone — CHANGES_REQUESTED** [review]
Looking good overall! Since it looks like the automation API doesn't support a few existing features yet, do we want to hold off on this, or is it better to move ahead with the refactor?

**Comment by EvanBoyle** [comment]
I believe the secrets provider work is the last big TODO for automation API and we should have feature parity after that. I'm not opposed to holding off until that lands.

**Comment by metral** [comment]
> do we want to hold off on this, or is it better to move ahead with the refactor?

> I'm not opposed to holding off until that lands.

Thanks for the review! Holding off SGTM 👍

**Review by lblackstone — APPROVED** [review]
🎉  Looks awesome!

**Review by EvanBoyle — APPROVED** [review]
This looks great! Awesome to see so much red! 🚀 

One thing I noticed was that there is a small regression in functionality. Previously `pulumi up` and company were being run through the `runCmd` func which has it's own io handling to pipe stdout to logs that can be followed by kubectl and such:

https://github.com/pulumi/pulumi-kubernetes-operator/blob/6a4943af8b888adcf3f7fc2b0400f57768a0dcf7/pkg/controller/stack/stack_controller.go#L500-L530

Given that updates are run through automation API, those logs won't flow anymore. A solution here is to wire up `ProgressStreams` from this PR https://github.com/pulumi/pulumi/pull/5367 for up/refresh/destroy/preview with the `io.Writer` from the logging interface (not sure if that's a logging interface or if it just goes to stdout). Not blocking, but something to consider.

File: pkg/controller/stack/stack_controller.go
Comment (metral) [reply pkg/controller/stack/stack_controller.go:520]: Ah good point! I'll move it before stack upsert for the time being until 5370 is addressed.

**Comment by metral** [comment]
Opened https://github.com/pulumi/pulumi-kubernetes-operator/issues/87 to track the use of `ProgressStreams` for logs.

**Pull request merged**
